vars/launchable.groovy

```diff
@@ -1,23 +1,43 @@
 def install() {
+  boolean alreadyInstalled = true
   if (isUnix()) {
     catchError(buildResult: 'SUCCESS', catchInterruptions: false, message: 'Failed to install Launchable; continuing build.') {
-      sh '''
-          python3 -m venv launchable
-          launchable/bin/pip --require-virtualenv --no-cache-dir install -U setuptools wheel
-          launchable/bin/pip --require-virtualenv --no-cache-dir install launchable
-         '''.stripIndent()
+      if (sh(script: 'command -v launchable', returnStatus: true) != 0) {
+        sh '''
+            python3 -m venv launchable
+            launchable/bin/pip --require-virtualenv --no-cache-dir install -U setuptools wheel
+            launchable/bin/pip --require-virtualenv --no-cache-dir install launchable
+            ln -s launchable/bin/launchable /urs/local/bin/launchable
+            '''.stripIndent()
+        alreadyInstalled = false
+      }
     }
   } else {
-    error 'Launchable installation is not yet implemented for Windows'
+    catchError(buildResult: 'SUCCESS', catchInterruptions: false, message: 'Failed to install Launchable; continuing build.') {
+      if (bat(script: 'python -m launchable -version 2>NUL || echo "NOT_INSTALLED"', returnStdout: true) == 'NOT_INSTALLED') {
+        bat '''
+            python.exe -m pip --no-cache-dir install --upgrade setuptools wheel pip
+            python.exe -m pip --no-cache-dir install launchable
+          '''.stripIndent()
+        alreadyInstalled = false
+      }
+    }
+  }
+  if (alreadyInstalled) {
+    echo 'DEPRECATION NOTICE: Launchable is already installed, no need to run "launchable.install"'
   }
 }
 
 def call(String args) {
   if (isUnix()) {
     catchError(buildResult: 'SUCCESS', catchInterruptions: false, message: 'Failed to run Launchable; continuing build.') {
-      sh 'launchable/bin/launchable ' + args
+      sh 'launchable ' + args
     }
   } else {
-    error 'The Launchable CLI is not yet implemented for Windows'
+    catchError(buildResult: 'SUCCESS', catchInterruptions: false, message: 'Failed to run Launchable; continuing build.') {
+      // Avoid logging the errors due to Nano server used in docker-inbound-agent
+      // TODO remove ' 2>NUL' when switching to Windows Server images instead of Nano Server images
+      bat 'python -m launchable ' + args + ' 2>NUL'
+    }
```

Comment:
```suggestion
    def launchableStderrFile = 'launchable.stderr
    try {
      // Avoid logging the errors due to Nano server used in docker-inbound-agent
      // TODO remove ' 2>NUL' when switching to Windows Server images instead of Nano Server images
      bat 'python -m launchable ' + args + ' 2>' + launchableStderrFile
    catch (err) {
        echo "Caught: ${err}"
        currentBuild.result = 'SUCCESS'
    finally {
      bat 'type ' + launchableStderrFile
    }
```

Reply:
https://www.jenkins.io/doc/pipeline/steps/workflow-basic-steps/

Reply:
As seen in 1:1: the goal would be to avoid masking error messages from launchable when it fails (but when it succeeeds , e.g. nominal case, then we don't want to see stderr)